elephant_vending_machine/views.py

```diff
@@ -48,17 +52,36 @@ def upload_image():
     will be returned
 
     Returns:
-        HTTP response 200 with payload 'File received' or
-        HTTP response 500 with payload 'No image file in request.'
-
+        HTTP response 200 with payload 'Success: Image saved.' or
+        HTTP response 400 with payload describing the issue with the request.
     """
 
     response = ""
-    if 'file' in request.files:
-        response = "File received."
+    response_code = 400
+    if 'file' not in request.files:
+        response = "Error with request: No file field in body of request."
     else:
-        response = "No image file in request."
-    return response
+        file = request.files['file']
+        if file.filename == '':
+            response = "Error with request: File field in body of response with no file present."
+        elif file and allowed_file(file.filename):
+            filename = secure_filename(file.filename)
+            file.save(os.path.join(APP.config['IMAGE_UPLOAD_FOLDER'], filename))
+            response = "Success: Image saved."
+            response_code = 200
+        else:
+            response = "Error with request: File extension not allowed."
+    return response, response_code
+
+def allowed_file(filename):
```

Comment:
Nit: easier to read if you put this above the upload_image() function since that function calls this one.